client/src/components/UserPage.tsx

```diff
@@ -71,22 +90,22 @@ const UserPage = ({ match }: RouteComponentProps<RouteInfo>) => {
   // update data store with the user's accounts
   useEffect(() => {
     getAccountsByUser(userId);
-  }, [getAccountsByUser, userId]);
+  }, [getAccountsByUser, userId, numOfItems]);
 
   useEffect(() => {
     setAccounts(accountsByUser[userId] || []);
   }, [accountsByUser, userId]);
 
-  // creates new link token upon new user or change in number of items
   useEffect(() => {
-    if (userId != null) {
-      generateLinkToken(userId, null); // itemId is null
+    // checks identity of user against identity/get data stored in accounts data
+    // only checks if identity has not already been verified.
+    if (accounts.length > 0 && identityCheck === false) {
```

Comment:
nit: i'd also change identityCheck here to something like `isIdentityChecked` or `isIdentityVerified`